src/sipsess/reply.c

```diff
@@ -103,7 +104,9 @@ int sipsess_reply_2xx(struct sipsess *sess, const struct sip_msg *msg,
 
 	sip_contact_set(&contact, sess->cuser, &msg->dst, msg->tp);
 
-	err = sip_treplyf(&sess->st, &reply->mb, sess->sip,
+	is_prack = !pl_strcmp(&msg->met, "PRACK");
+
+	err = sip_treplyf(is_prack ? NULL : &sess->st, &reply->mb, sess->sip,
```

Comment:
In `sip/strans.c:sip_strans_reply:390` the dereferencing happens unconditionally after 64*T1 if the transport protocol used is not reliable. Otherwise it is derefed immediately. I am assuming there is a reason why that is happening.

Either way, dereferencing is guaranteed. Either immediately or after 64*T1 (for all non-INVITE server transactions).